R/profvis.R

```diff
@@ -140,7 +143,13 @@ print.profvis <- function(x, ..., width = NULL, height = NULL,
   if (viewer) {
     getS3method("print", "htmlwidget")(x)
   } else {
-    NextMethod()
+    f <- getOption("profvis.print")
+    if (is.function(f)) {
+      f(x)
```

Comment:
The `viewer=TRUE` path uses `getS3method("print", "htmlwidget")(x)` without any additional parameters so I'm inclined to keep the code as is. We will probably have enough customization functionality with only the results since the structure it returns under `x$message` is rich, here is a sample output:

```
$ x            :List of 1
  ..$ message:List of 6
  .. ..$ prof       :'data.frame':	1041 obs. of  6 variables:
  .. .. ..$ time    : int [1:1041] 1 1 1 1 1 1 1 1 1 1 ...
  .. .. ..$ depth   : int [1:1041] 12 11 10 9 8 7 6 5 4 3 ...
  .. .. ..$ label   : chr [1:1041] ".Call" "<Anonymous>" "myf" "force" ...
  .. .. ..$ filenum : int [1:1041] NA 1 NA NA NA NA NA NA NA NA ...
  .. .. ..$ linenum : int [1:1041] NA 3 NA NA NA NA NA NA NA NA ...
  .. .. ..$ filename: chr [1:1041] NA "~/RStudio/temp/recursive_func.R" NA NA ...
  .. ..$ interval   : num 10
  .. ..$ files      :List of 1
  .. .. ..$ :List of 2
  .. .. .. ..$ filename: chr "~/RStudio/temp/recursive_func.R"
  .. .. .. ..$ content : chr "myf <- function(e) {\n  if (e > 0) {\n    profvis::pause(0.2)\n    myf(e-1)\n  }\n}\nmyf(5)\n"
  .. ..$ prof_output: chr "/Users/javierluraschi/RStudio/profvis/.Rproj.user/90639BFA/profiles-cache/file74815bb9b05d.rprof"
  .. ..$ highlight  :List of 3
  .. .. ..$ output    :List of 1
  .. .. .. ..$ : chr "^output\\$"
  .. .. ..$ gc        :List of 1
  .. .. .. ..$ : chr "^<GC>$"
  .. .. ..$ stacktrace:List of 1
  .. .. .. ..$ : chr "^\\.\\.stacktraceo(n|ff)\\.\\.$"
  .. ..$ split      : chr "v"
 $ width        : NULL
 $ height       : NULL
 $ sizingPolicy :List of 6
  ..$ defaultWidth : NULL
  ..$ defaultHeight: NULL
  ..$ padding      : num 0
  ..$ viewer       :List of 6
  .. ..$ defaultWidth : NULL
  .. ..$ defaultHeight: NULL
  .. ..$ padding      : NULL
  .. ..$ fill         : logi TRUE
  .. ..$ suppress     : logi TRUE
  .. ..$ paneHeight   : NULL
  ..$ browser      :List of 4
  .. ..$ defaultWidth : NULL
  .. ..$ defaultHeight: NULL
  .. ..$ padding      : NULL
  .. ..$ fill         : logi TRUE
  ..$ knitr        :List of 3
  .. ..$ defaultWidth : chr "100%"
  .. ..$ defaultHeight: chr "600px"
  .. ..$ figure       : logi FALSE
 $ dependencies : NULL
 $ elementId    : NULL
 $ preRenderHook: NULL
 - attr(*, "class")= chr [1:3] "profvis" "suppress_viewer" "htmlwidget"
 - attr(*, "package")= chr "profvis"
```